app/clients/sms/aws_sns.py

```diff
@@ -56,3 +82,6 @@ def send_sms(self, to, content, reference, multi=True, sender=None):
             self.statsd_client.incr("clients.sns.error")
             self.current_app.logger.error("No valid numbers found in {}".format(to))
             raise ValueError("No valid numbers found for SMS delivery")
+
+    def _send_with_dedicated_phone_number(self, sender):
+        return sender and re.match(self.long_code_regex, sender)
```

Comment:
On the `Notification` model, the check is only `sender[0] == "+"` but I've been extra safe here

Reply:
not a huge deal, but the naming of sender is ambiguous at first glance, it sounds like the entity/service doing the sending and not the 'send as number'.

Reply:
Yeah, agreed. I didn't pick this name. `sender` but be understood as "the number sending the text message".

The call to this method is here https://github.com/cds-snc/notification-api/blob/a35f8fe77bc9f5a7fd8b32f567ee720ce241d091/app/delivery/send_to_providers.py#L73-L78 and other providers adhere to this interface